### PR TITLE
Add option to save logit scores for inference tif

### PR DESCRIPTION
**Pull request by hannah-rae**
Currently the CLI only has the option to save the segmentation class output (3 classes, 1 channel output raster). 

I want the option to save the raw logit prediction scores before the argmax is applied to get the predicted class in each pixel (3 classes, 3 channel output raster). For example, the scores are needed to use Lyndon Estes' [instancemaker](https://github.com/agroimpacts/instancemaker) library for fancier polygonizing than we have in the repo currently. 

This PR adds the command line option `--save_scores` which saves the prediction scores instead of the classes. 

Example output for Austria:
```ftw inference all \                                              
    --bbox=13.0,48.0,13.2,48.2 \
    --year=2024 \
    --out=test-austria-all \
    --cloud_cover_max=20 \
    --buffer_days=14 \
    --model=model-ckpts/3_Class_FULL_FTW_Pretrained.ckpt \
    --resize_factor=2 \
    --overwrite \
    --save_scores
```
image

```ftw inference all \                                              
    --bbox=13.0,48.0,13.2,48.2 \
    --year=2024 \
    --out=test-austria-all \
    --cloud_cover_max=20 \
    --buffer_days=14 \
    --model=model-ckpts/3_Class_FULL_FTW_Pretrained.ckpt \
    --resize_factor=2 \
    --overwrite
```
image

Note: I think the stripes on the sides are due to [this issue](https://github.com/fieldsoftheworld/ftw-inference-app/issues/73) which @calebrob6 addressed in #166

### CODE REVIEW

**Comment by calebrob6** [comment]
LGTM -- only potential change I see is computing the softmax over the model output (currently outputs logits). This lets you:
- quantize to [0,255] so you can save the scores as uint8 which will make the file size much smaller
- interpret scores directly as probabilities

**Comment by hannah-rae** [comment]
Good suggestion! I thought about that, but wasn't sure if it would be confusing to people by making them think we were using softmax internally. Either way that's not a big risk and the benefit of storing as uint8 is huge. I'll make the change today!

**Review by calebrob6 — APPROVED** [review]
